internal/commands/pull/inputs.go

```diff
@@ -109,3 +117,61 @@ func (i *inputs) resolveRemoteApp(ui terminal.UI, clients cli.Clients) (realm.Ap
 
 	return app, nil
 }
+
+func (i *inputs) resolveClient(ui terminal.UI, realmClient realm.Client, groupID, appID string) (map[string]*zip.Reader, bool, error) {
+	compatibleTemplates, err := realmClient.CompatibleTemplates(groupID, appID)
```

Comment:
i think this was where nick was trying to add the `bool` return param.

I think this should be something like

```go
// its possible that the request was successful, but there was no template for the current app.
// checking ok allows us to have more granular context for what went wrong
compatibleTemplates, ok, err := realmClient.CompatibleTemplates(groupID, appID)
if err != nil {
	// something went wrong
}
if !ok {
	// request succesful, but nothing returned
	return nil
}
// client exists so we handle it
```